### ISSUE TITLE
undefined method `reorder' for nil:NilClass

### ISSUE THREAD

**Issue by ain** [activeadmin/activeadmin#3733]
```
/Users/ain/.rvm/gems/ruby-2.2.0/gems/activeadmin-0.6.3/app/views/active_admin/resource/index.html.arb where line #2 raised:

undefined method `reorder' for nil:NilClass
```

Request:
```
{"order"=>"updated_at_desc"}
```

The code:
```ruby
config.sort_order = "updated_at_desc"
```

Full stack:
```
activeadmin (0.6.3) lib/active_admin/helpers/collection.rb:11:in `collection_size'
activeadmin (0.6.3) lib/active_admin/helpers/collection.rb:19:in `collection_is_empty?'
activeadmin (0.6.3) lib/active_admin/views/pages/index.rb:44:in `items_in_collection?'
activeadmin (0.6.3) lib/active_admin/views/pages/index.rb:48:in `build_collection'
activeadmin (0.6.3) lib/active_admin/views/pages/index.rb:27:in `block in main_content'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:31:in `block in build_tag'
arbre (1.0.2) lib/arbre/context.rb:92:in `with_current_arbre_element'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:49:in `with_current_arbre_element'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:26:in `build_tag'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:39:in `insert_tag'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:14:in `batch_action_form'
activeadmin (0.6.3) lib/active_admin/views/pages/index.rb:35:in `wrap_with_batch_action_form'
activeadmin (0.6.3) lib/active_admin/views/pages/index.rb:25:in `main_content'
activeadmin (0.6.3) lib/active_admin/views/pages/base.rb:83:in `block (2 levels) in build_main_content_wrapper'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:31:in `block in build_tag'
arbre (1.0.2) lib/arbre/context.rb:92:in `with_current_arbre_element'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:49:in `with_current_arbre_element'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:26:in `build_tag'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:39:in `insert_tag'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:14:in `div'
activeadmin (0.6.3) lib/active_admin/views/pages/base.rb:82:in `block in build_main_content_wrapper'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:31:in `block in build_tag'
arbre (1.0.2) lib/arbre/context.rb:92:in `with_current_arbre_element'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:49:in `with_current_arbre_element'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:26:in `build_tag'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:39:in `insert_tag'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:14:in `div'
activeadmin (0.6.3) lib/active_admin/views/pages/base.rb:81:in `build_main_content_wrapper'
activeadmin (0.6.3) lib/active_admin/views/pages/base.rb:65:in `block in build_page_content'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:31:in `block in build_tag'
arbre (1.0.2) lib/arbre/context.rb:92:in `with_current_arbre_element'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:49:in `with_current_arbre_element'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:26:in `build_tag'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:39:in `insert_tag'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:14:in `div'
activeadmin (0.6.3) lib/active_admin/views/pages/base.rb:64:in `build_page_content'
activeadmin (0.6.3) lib/active_admin/views/pages/base.rb:47:in `block (2 levels) in build_page'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:31:in `block in build_tag'
arbre (1.0.2) lib/arbre/context.rb:92:in `with_current_arbre_element'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:49:in `with_current_arbre_element'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:26:in `build_tag'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:39:in `insert_tag'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:14:in `div'
activeadmin (0.6.3) lib/active_admin/views/pages/base.rb:44:in `block in build_page'
arbre (1.0.2) lib/arbre/context.rb:92:in `with_current_arbre_element'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:49:in `with_current_arbre_element'
activeadmin (0.6.3) lib/active_admin/views/pages/base.rb:43:in `build_page'
activeadmin (0.6.3) lib/active_admin/views/pages/base.rb:10:in `build'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:30:in `block in build_tag'
arbre (1.0.2) lib/arbre/context.rb:92:in `with_current_arbre_element'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:26:in `build_tag'
arbre (1.0.2) lib/arbre/element/builder_methods.rb:39:in `insert_tag'
activeadmin (0.6.3) app/views/active_admin/resource/index.html.arb:2:in `block in ___sers_ain__rvm_gems_ruby_______gems_activeadmin_______app_views_active_admin_resource_index_html_arb__3862692562881753141_70138334420400'
arbre (1.0.2) lib/arbre/context.rb:45:in `instance_eval'
arbre (1.0.2) lib/arbre/context.rb:45:in `initialize'
activeadmin (0.6.3) app/views/active_admin/resource/index.html.arb:1:in `new'
activeadmin (0.6.3) app/views/active_admin/resource/index.html.arb:1:in `___sers_ain__rvm_gems_ruby_______gems_activeadmin_______app_views_active_admin_resource_index_html_arb__3862692562881753141_70138334420400'
actionpack (3.2.21) lib/action_view/template.rb:145:in `block in render'
activesupport (3.2.21) lib/active_support/notifications.rb:125:in `instrument'
actionpack (3.2.21) lib/action_view/template.rb:143:in `render'
actionpack (3.2.21) lib/action_view/renderer/template_renderer.rb:47:in `block (2 levels) in render_template'
actionpack (3.2.21) lib/action_view/renderer/abstract_renderer.rb:38:in `block in instrument'
activesupport (3.2.21) lib/active_support/notifications.rb:123:in `block in instrument'
activesupport (3.2.21) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
activesupport (3.2.21) lib/active_support/notifications.rb:123:in `instrument'
actionpack (3.2.21) lib/action_view/renderer/abstract_renderer.rb:38:in `instrument'
actionpack (3.2.21) lib/action_view/renderer/template_renderer.rb:46:in `block in render_template'
actionpack (3.2.21) lib/action_view/renderer/template_renderer.rb:54:in `render_with_layout'
actionpack (3.2.21) lib/action_view/renderer/template_renderer.rb:45:in `render_template'
actionpack (3.2.21) lib/action_view/renderer/template_renderer.rb:18:in `render'
actionpack (3.2.21) lib/action_view/renderer/renderer.rb:36:in `render_template'
actionpack (3.2.21) lib/action_view/renderer/renderer.rb:17:in `render'
actionpack (3.2.21) lib/abstract_controller/rendering.rb:110:in `_render_template'
actionpack (3.2.21) lib/action_controller/metal/streaming.rb:225:in `_render_template'
actionpack (3.2.21) lib/abstract_controller/rendering.rb:103:in `render_to_body'
actionpack (3.2.21) lib/action_controller/metal/renderers.rb:28:in `render_to_body'
actionpack (3.2.21) lib/action_controller/metal/compatibility.rb:50:in `render_to_body'
actionpack (3.2.21) lib/abstract_controller/rendering.rb:88:in `render'
actionpack (3.2.21) lib/action_controller/metal/rendering.rb:16:in `render'
actionpack (3.2.21) lib/action_controller/metal/instrumentation.rb:40:in `block (2 levels) in render'
activesupport (3.2.21) lib/active_support/core_ext/benchmark.rb:5:in `block in ms'
/Users/ain/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
activesupport (3.2.21) lib/active_support/core_ext/benchmark.rb:5:in `ms'
actionpack (3.2.21) lib/action_controller/metal/instrumentation.rb:40:in `block in render'
actionpack (3.2.21) lib/action_controller/metal/instrumentation.rb:83:in `cleanup_view_runtime'
activerecord (3.2.21) lib/active_record/railties/controller_runtime.rb:24:in `cleanup_view_runtime'
actionpack (3.2.21) lib/action_controller/metal/instrumentation.rb:39:in `render'
activeadmin (0.6.3) lib/active_admin/resource_controller/actions.rb:13:in `block (2 levels) in index'
actionpack (3.2.21) lib/action_controller/metal/responder.rb:230:in `call'
actionpack (3.2.21) lib/action_controller/metal/responder.rb:230:in `default_render'
actionpack (3.2.21) lib/action_controller/metal/responder.rb:160:in `to_html'
responders (1.1.2) lib/responders/flash_responder.rb:107:in `to_html'
actionpack (3.2.21) lib/action_controller/metal/responder.rb:153:in `respond'
actionpack (3.2.21) lib/action_controller/metal/responder.rb:146:in `call'
actionpack (3.2.21) lib/action_controller/metal/mime_responds.rb:239:in `respond_with'
inherited_resources (1.5.1) lib/inherited_resources/actions.rb:7:in `index'
activeadmin (0.6.3) lib/active_admin/resource_controller/actions.rb:11:in `index'
activeadmin-axlsx (2.1.2) lib/active_admin/axlsx/resource_controller_extension.rb:12:in `index_with_xlsx'
actionpack (3.2.21) lib/action_controller/metal/implicit_render.rb:4:in `send_action'
actionpack (3.2.21) lib/abstract_controller/base.rb:167:in `process_action'
actionpack (3.2.21) lib/action_controller/metal/rendering.rb:10:in `process_action'
actionpack (3.2.21) lib/abstract_controller/callbacks.rb:18:in `block in process_action'
activesupport (3.2.21) lib/active_support/callbacks.rb:458:in `_run__3563225836772307296__process_action__2636623223988250929__callbacks'
activesupport (3.2.21) lib/active_support/callbacks.rb:405:in `__run_callback'
activesupport (3.2.21) lib/active_support/callbacks.rb:385:in `_run_process_action_callbacks'
activesupport (3.2.21) lib/active_support/callbacks.rb:81:in `run_callbacks'
actionpack (3.2.21) lib/abstract_controller/callbacks.rb:17:in `process_action'
actionpack (3.2.21) lib/action_controller/metal/rescue.rb:29:in `process_action'
actionpack (3.2.21) lib/action_controller/metal/instrumentation.rb:30:in `block in process_action'
activesupport (3.2.21) lib/active_support/notifications.rb:123:in `block in instrument'
activesupport (3.2.21) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
activesupport (3.2.21) lib/active_support/notifications.rb:123:in `instrument'
actionpack (3.2.21) lib/action_controller/metal/instrumentation.rb:29:in `process_action'
actionpack (3.2.21) lib/action_controller/metal/params_wrapper.rb:207:in `process_action'
activerecord (3.2.21) lib/active_record/railties/controller_runtime.rb:18:in `process_action'
actionpack (3.2.21) lib/abstract_controller/base.rb:121:in `process'
actionpack (3.2.21) lib/abstract_controller/rendering.rb:45:in `process'
actionpack (3.2.21) lib/action_controller/metal.rb:203:in `dispatch'
actionpack (3.2.21) lib/action_controller/metal/rack_delegation.rb:14:in `dispatch'
actionpack (3.2.21) lib/action_controller/metal.rb:246:in `block in action'
actionpack (3.2.21) lib/action_dispatch/routing/route_set.rb:73:in `call'
actionpack (3.2.21) lib/action_dispatch/routing/route_set.rb:73:in `dispatch'
actionpack (3.2.21) lib/action_dispatch/routing/route_set.rb:36:in `call'
journey (1.0.4) lib/journey/router.rb:68:in `block in call'
journey (1.0.4) lib/journey/router.rb:56:in `each'
journey (1.0.4) lib/journey/router.rb:56:in `call'
actionpack (3.2.21) lib/action_dispatch/routing/route_set.rb:608:in `call'
warden (1.2.3) lib/warden/manager.rb:35:in `block in call'
warden (1.2.3) lib/warden/manager.rb:34:in `catch'
warden (1.2.3) lib/warden/manager.rb:34:in `call'
actionpack (3.2.21) lib/action_dispatch/middleware/best_standards_support.rb:17:in `call'
rack (1.4.5) lib/rack/etag.rb:23:in `call'
rack (1.4.5) lib/rack/conditionalget.rb:25:in `call'
actionpack (3.2.21) lib/action_dispatch/middleware/head.rb:14:in `call'
actionpack (3.2.21) lib/action_dispatch/middleware/params_parser.rb:21:in `call'
actionpack (3.2.21) lib/action_dispatch/middleware/flash.rb:242:in `call'
rack (1.4.5) lib/rack/session/abstract/id.rb:210:in `context'
rack (1.4.5) lib/rack/session/abstract/id.rb:205:in `call'
actionpack (3.2.21) lib/action_dispatch/middleware/cookies.rb:341:in `call'
activerecord (3.2.21) lib/active_record/query_cache.rb:64:in `call'
activerecord (3.2.21) lib/active_record/connection_adapters/abstract/connection_pool.rb:479:in `call'
actionpack (3.2.21) lib/action_dispatch/middleware/callbacks.rb:28:in `block in call'
activesupport (3.2.21) lib/active_support/callbacks.rb:405:in `_run__3127652035251521944__call__829184060372439226__callbacks'
activesupport (3.2.21) lib/active_support/callbacks.rb:405:in `__run_callback'
activesupport (3.2.21) lib/active_support/callbacks.rb:385:in `_run_call_callbacks'
activesupport (3.2.21) lib/active_support/callbacks.rb:81:in `run_callbacks'
actionpack (3.2.21) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
actionpack (3.2.21) lib/action_dispatch/middleware/reloader.rb:65:in `call'
actionpack (3.2.21) lib/action_dispatch/middleware/remote_ip.rb:31:in `call'
actionpack (3.2.21) lib/action_dispatch/middleware/debug_exceptions.rb:16:in `call'
actionpack (3.2.21) lib/action_dispatch/middleware/show_exceptions.rb:56:in `call'
railties (3.2.21) lib/rails/rack/logger.rb:32:in `call_app'
railties (3.2.21) lib/rails/rack/logger.rb:16:in `block in call'
activesupport (3.2.21) lib/active_support/tagged_logging.rb:22:in `tagged'
railties (3.2.21) lib/rails/rack/logger.rb:16:in `call'
actionpack (3.2.21) lib/action_dispatch/middleware/request_id.rb:22:in `call'
rack (1.4.5) lib/rack/methodoverride.rb:21:in `call'
rack (1.4.5) lib/rack/runtime.rb:17:in `call'
activesupport (3.2.21) lib/active_support/cache/strategy/local_cache.rb:72:in `call'
rack (1.4.5) lib/rack/lock.rb:15:in `call'
actionpack (3.2.21) lib/action_dispatch/middleware/static.rb:83:in `call'
railties (3.2.21) lib/rails/engine.rb:484:in `call'
railties (3.2.21) lib/rails/application.rb:231:in `call'
rack (1.4.5) lib/rack/content_length.rb:14:in `call'
railties (3.2.21) lib/rails/rack/log_tailer.rb:17:in `call'
rack (1.4.5) lib/rack/handler/webrick.rb:59:in `service'
/Users/ain/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/webrick/httpserver.rb:138:in `service'
/Users/ain/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/webrick/httpserver.rb:94:in `run'
/Users/ain/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/webrick/server.rb:294:in `block in start_thread'
```

**Comment by timoschilling** [comment]
Rails 3.2 don't work well with Ruby 2.2.

Do you have the same problem with Ruby 2.1?

Please post your model and your active admin resource.

**Comment by ain** [comment]
Ruby 2.1 works fine, true, but IMO there's quite a performance difference
in favour of 2.2 which is why I brought up the ticket.

I have 2.2 working in production with Rails 3.2 since it was released now,
no problems so far.

**Comment by timoschilling** [comment]
Sure 2.2 is nice, but 3.2 don't work with 2.2 at the moment. See the failing tests https://travis-ci.org/activeadmin/activeadmin/builds/46036663 and #3715.

I'm closing this as a double of #3715, if #3715 is closed and the problem still exist, ping me and I will reopen the issue.

**Comment by timoschilling** [comment]
@ain do you still have this problem with the latest master?

**Comment by ain** [comment]
Haven't tried recently TBH, but could give it a run on my staging late this
week/beginning next week :)

-- 
Sent from Gmail Mobile

**Comment by timoschilling** [comment]
@ain closed by #3760?
